src/bluesearch/database/topic.py

```diff
@@ -295,3 +297,63 @@ def get_topics_for_pmc_article(
             journal_topics.append(descriptor["name"])
 
     return journal_topics
+
+
+def extract_info_from_zipfile(path: pathlib.Path) -> tuple[str, str]:
+    """Extract topic of a medRxiv/bioRxiv article.
+
+    The `.meca` file should always have a fixed structure. Namely,
+    there is a folder `content` and inside of it there should be
+    a single `.xml` file containing the text and the metadata of the
+    article.
+
+    Parameters
+    ----------
+    path
+        Path to a `.meca` file (which is nothing else
+        than a zip archive) with a fixed structured.
+
+    Returns
+    -------
+    topic : str
+        The subject area of the article.
+    journal : str
+        The journal the article was published in. Should be either
+        "medRxiv" or "bioRxiv".
+
+    Raises
+    ------
+    ValueError
+        Appropriate XML not found or the journal or topic are missing.
+    """
+    with zipfile.ZipFile(path) as myzip:
+        xml_files = [
+            x
+            for x in myzip.namelist()
+            if x.startswith("content/") and x.endswith(".xml")
+        ]
+
+        if len(xml_files) != 1:
+            raise ValueError(
+                "There needs to be exactly one .xml file inside of content/"
+            )
+
+        xml_file = xml_files[0]
+        pattern_topic = """<subj-group subj-group-type="hwp-journal-coll">\r\n<subject>(.*)</subject>"""  # noqa
+        pattern_journal = "<journal-title>(.*)</journal-title>"
+
+        with myzip.open(xml_file, "r") as f:
+            text = f.read().decode("utf-8")
+            match_topic = re.search(pattern_topic, text)
+            match_journal = re.search(pattern_journal, text)
+
+            if match_topic is None:
+                raise ValueError("No topic found")
+
+            if match_journal is None:
+                raise ValueError("No journal found")
+
+            topic = match_topic.groups()[0]
+            journal = match_journal.groups()[0]
```

Comment:
Instead of `groups()[0]` one can write `group(0)`.

Reply:
It should be `group(1)` right (unit tests didn't like it)? https://docs.python.org/3/library/re.html#re.Match.group

Anyway, done in d7bddd0

Reply:
Ah good point!